style/hudl_sql_style_guide.md

```diff
@@ -0,0 +1,339 @@
+JOIN ksr.users AS users ON ...
+LEFT JOIN ...
+JOIN ksr.locations AS locations ON ...
+```
+
+## `WHERE`
+
+Multiple `WHERE` clauses should go on different lines and begin with the SQL operator:
+
+```sql
+SELECT
+  name,
+  goal
+FROM ksr.projects AS projects
+WHERE
+  country = 'US'
+  AND deadline >= '2015-01-01'
+...
+```
+
+## `GROUP BY`
+
+GROUP BY/ORDER BY (group by and order by on line by itself, list of fields start on next line 2 spaces in. 
+
+```SQL
+GROUP BY 
+  schoolid, 
+  name, 
+  organization_type
+```
+
+
+## `CASE`
+
+`CASE` statements aren't always easy to format but try to align `WHEN` and `ELSE` together inside `CASE` and `END`:
+
+```sql
+CASE 
+  WHEN category = 'Art' THEN backer_id
+  ELSE NULL
+END
+```
+
+## Common Table Expressions (CTEs)
+
+[From AWS](http://docs.aws.amazon.com/redshift/latest/dg/r_WITH_clause.html):
+
+>`WITH` clause subqueries are an efficient way of defining tables that can be used throughout the execution of a single query. In all cases, the same results can be achieved by using subqueries in the main body of the `SELECT` statement, but `WITH` clause subqueries may be simpler to write and read.
+
+The body of a CTE must be one indent further than the `WITH` keyword. Open them at the end of a line and close them on a new line:
+
+```sql
+WITH backings_per_category AS (
+  SELECT
+    category_id,
+    deadline,
+    ...
+)
+```
+
+Multiple CTEs should be formatted accordingly:
+
+```sql
+WITH backings_per_category AS (
+  SELECT
+    ...
+), backers AS (
+  SELECT
+    ...
+), backers_and_creators AS (
+  ...
+)
+SELECT * FROM backers;
+```
+
+If possible, `JOIN` CTEs inside subsequent CTEs, not in the main clause:
+
+__GOOD__:
+
+```sql
+WITH backings_per_category AS (
+  SELECT
+    ...
+), backers AS (
+  SELECT
+    backer_id,
+    COUNT(backings_per_category.id) AS projects_backed_per_category
+  JOIN ksr.users AS users ON users.id = backings_per_category.backer_id
+), backers_and_creators AS (
+  ...
+)
+SELECT * FROM backers_and_creators;
+```
+
+__BAD__:
+
+```sql
+WITH backings_per_category AS (
+  SELECT
+    ...
+), backers AS (
+  SELECT
+    backer_id,
+    COUNT(backings_per_category.id) AS projects_backed_per_category
+), backers_and_creators AS (
+  ...
+)
+SELECT * FROM backers_and_creators
+JOIN backers ON backers_and_creators ON backers.backer_id = backers_and_creators.backer_id
+```
+
+Always use CTEs over inlined subqueries.
+
+## Comments
+Include comments for any code that isn’t obvious
+Comments that are one line can be done with double dash
+
+```SQL
+--This is a comment
+```
+
+Comments that are longer and exceed 80 character limit should be down using /* as seen below
+
+```SQL
+/*
+Large block of comments
+*/
+```
+
+Don’t commit code to master with code commented out unless we have a good reason for it
+
+Use Ramzi style Dividers between CTEs, subqueries
+https://github.com/hudl/data-redshift-maintenance/blob/master/src/dbt/models/performance/atrest/tables.sql#L25 
+
```

Comment:
Is this a requirement or a suggestion? It might be useful some places, but as a rule I'm not sure I agree.

Reply:
Good question. I copied that over from our style guide. I'm open to suggestions.